### DISCUSSION TITLE
Documentation / Example of debug adapter IDE integration?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse/lsp4e#931] [Q&A]
I'm currently looking into https://github.com/eclipse/lsp4e/tree/master/org.eclipse.lsp4e.debug and the examples seem only contain how to _launch_ a debug adapter but I suppose lsp4e won't discover that automatically (or will it?) so I'm wonder what needs to be done minimally to integrate such an adapter that it show up in the IDE.

**Comment by mickaelistria** [comment]
You'll basically need to create a new launch config (and its UI) that extends the LSP4E DAP launch config, and have the various settings of DAP launch set according to user input. Then this launch will call the DAP launch which in turn will launch the "target" and report events through DAP.
